sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DeduplicateRelations.scala

```diff
@@ -105,6 +105,21 @@ object DeduplicateRelations extends Rule[LogicalPlan] {
         (m, false)
       }
 
+    case p @ Project(_, child) if p.resolved && p.projectList.forall(_.isInstanceOf[Alias]) =>
+      val (renewed, changed) = renewDuplicatedRelations(existingRelations, child)
+      val (newPlan, planChanged) = getNewPlanWithNewChildren(existingRelations, Array(renewed),
+        p, changed)
+      var newProject = newPlan.asInstanceOf[Project]
+      val planWrapper = RelationWrapper(p.getClass, newProject.projectList.map(_.exprId.id))
```

Comment:
`Project` is a common node and let's not always create the `RelationWrapper` here.

We can find the new attributes precisely from `Project`:
```
projectList.collect {
  case a: Alias => a.exprId.id
}
```